MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs

```diff
@@ -250,18 +258,34 @@ protected void ParseProviderLinks(T item, string xml)
             }
         }
 
-        protected virtual void FetchDataFromXmlNode(XmlReader reader, MetadataResult<T> itemResult)
+        protected virtual XbmcMetadataOptions RetrieveNfoConfiguration()
         {
-            var item = itemResult.Item;
+            if (_nfoConfiguration == null)
+            {
+                _nfoConfiguration = _config.GetNfoConfiguration();
+            }
 
-            var nfoConfiguration = _config.GetNfoConfiguration();
-            UserItemData? userData = null;
-            if (!string.IsNullOrWhiteSpace(nfoConfiguration.UserId))
+            return _nfoConfiguration;
+        }
+
+        protected virtual UserItemData? RetrieveUserData(MetadataResult<T> itemResult)
+        {
+            if (_userData == null)
             {
-                var user = _userManager.GetUserById(Guid.Parse(nfoConfiguration.UserId));
-                userData = _userDataManager.GetUserData(user, item);
+                var nfoConfiguration = RetrieveNfoConfiguration();
+                if (!string.IsNullOrWhiteSpace(nfoConfiguration.UserId))
+                {
+                    _userData = itemResult.GetOrAddUserData(new Guid(nfoConfiguration.UserId));
+                }
```

Comment:
```suggestion
                if (Guid.TryParse(nfoConfiguration.UserId, out var userId))
                {
                    _userData = itemResult.GetOrAddUserData(userId);
                }
```

Reply:
Sure you want to silently ignore invalid values for UserId in nfoConfiguration, instead of signaling an error in that case? I don't know the proceedings in this project, but it sounds weird to me to ignore errors ... I you say 'yes', I will do.

Reply:
Crashing is not much better. Logging a suitable warning and moving on is better.